R/backport_linter.R

```diff
@@ -93,7 +93,11 @@ normalize_r_version <- function(r_version) {
     version_names <- c("devel", "release", paste0("oldrel-", seq_len(length(minor_versions) - 2L)))
     if (!r_version %in% version_names) {
       # This can only trip if e.g. oldrel-99 is requested
-      stop("`r_version` must be a version number or one of ", toString(sQuote(version_names)), call. = FALSE)
+      cli_abort(c(
+        "{.arg r_version} is not valid:",
+        i = "It must be a version number or one of {sQuote(version_names)}.",
```

Comment:
how does {cli} handle aggregation? before we needed `toString()` for the vector `version_names`.

Reply:
PS I'm not generally a fan of `sQuote()` since it's inconsistent across platforms, WDYT about making this `{.val r_version}` instead?

Reply:
I don't like it either, but I wanted to preserve the previous behaviour. Happy to change that.

Just one thing to keep in mind: Although the value is highlighted in the console and so it stands out:

```r
r_version <- getRversion()

f <- \() cli::cli_abort("Version {.val {r_version}} is not valid")
g <- \() cli::cli_abort("Version {.val {sQuote(r_version)}} is not valid")

f()
g()
```

<img width="486" alt="Screenshot 2024-04-13 at 15 57 53" src="https://github.com/r-lib/lintr/assets/11330453/fde3ca57-1aa7-47b8-b4e5-fb156c92f0b8">

That is not the case when we need to write unit tests:

```r
expect_error(f(), "Version 4.3.3 is not valid")
expect_error(g(), "Version '4.3.3' is not valid")
```

But I don't think it's too big a deal.

Reply:
```r
f <- \() cli::cli_abort("Version {.str {r_version}} is not valid")
```

would quote it!

Reply:
> I wanted to preserve the previous behaviour.

Here's a case where I'm happy to break with old behavior, part of the advantage of moving to {cli} is easier consistency on things like this where we can get the same formatting of arguments, function names, packages etc. by off-loading that to {cli}. Things like `sQuote()` I see as our piecemeal attempt at this kind of formatting, and can be abandoned now.